tests/test_foundation/test_network_types.py

```diff
@@ -15,3 +15,12 @@ def test_GenTLV_value_unpack(self):
         tlv_unpacked = GenericTLV()
         tlv_unpacked.unpack(tlv.pack())
         self.assertEqual(tlv.value.value, tlv_unpacked.value.value)
+
+    def test_IPv4_pack_unpack(self):
+        """Test pack/unpack of IPv4 class."""
+        packet = IPv4(ttl=64, protocol=17, source="192.168.0.1",
+                      destination="172.16.200.132", data=b'testdata')
+        packed = packet.pack()
+        unpacked = IPv4()
+        unpacked.unpack(packed)
+        self.assertEqual(packed, unpacked.pack())
```

Comment:
It is not a good way to test our pack/unpack against each other. I mean, the result of the pack method should be tested against the known expected result, while the result of the unpack method should be testes against the known expected result for the unpack.